ghmirror/core/mirror_requests.py

```diff
@@ -118,49 +181,40 @@ def online_request(method, url, auth, data=None, url_params=None):
         if last_mod is not None:
             headers['If-Modified-Since'] = last_mod
 
-    resp = requests.request(method=method,
-                            url=url,
-                            headers=headers,
-                            timeout=REQUESTS_TIMEOUT,
-                            params=parameters)
+    resp = _online_request(method=method,
+                           url=url,
+                           headers=headers,
+                           parameters=parameters,
+                           cached_response=cached_response)
 
     if resp.status_code == 304:
-        if len(cached_response.json()) == per_page_elements and \
-           not cached_response.links:
+        return _handle_not_changed(cached_response, per_page_elements,
+                                   headers, method, url, parameters,
+                                   cache, cache_key)
```

Comment:
Moved the logic to decide if we should make a new request inside, to be able to return the response directly from the function instead of checking the output and returning in the "main", I understand  this as part of this PR, but I can split this into another PR

Reply:
ok, leave it here